### PR TITLE
[Metricbeat]Test kube-state-metrics modules using v1.8

### PR DESCRIPTION
**Pull request by odacremolbap**
Test all kubernetes kube-state-metrics metricsets using v1.8

v1.3 is kept but we are guarantee no compatibility with that version.
Some metricset include extra ksm resources made up from ksm unit test data.

Fixes https://github.com/elastic/beats/issues/13986

### CODE REVIEW

File: metricbeat/module/kubernetes/state_persistentvolumeclaim/state_persistentvolumeclaim_test.go [hunk 1]
```diff
@@ -28,6 +28,14 @@ import (
...
+			{
+				MetricsFile:  "../_meta/test/ksm.v1.3.0",
+				ExpectedFile: "./_meta/test/ksm.v1.3.0.expected",
+			},
+			{
+				MetricsFile:  "../_meta/test/ksm.v1.8.0",
+				ExpectedFile: "./_meta/test/ksm.v1.8.0.expected",
+			},
 			{
 				MetricsFile:  "./_meta/test/ksm-unit-1.8",
```
Comment (ChrsMark) [inline metricbeat/module/kubernetes/state_persistentvolumeclaim/state_persistentvolumeclaim_test.go]: Would it be of benefit to keep some consistency here regarding the naming?:
```suggestion
				MetricsFile:  "./_meta/test/ksm.unit.v1.8.0",
```

Comment (odacremolbap) [reply metricbeat/module/kubernetes/state_persistentvolumeclaim/state_persistentvolumeclaim_test.go]: sounds good

**Comment by odacremolbap** [comment]
done @ChrsMark

**Comment by ChrsMark** [comment]
> done @ChrsMark

Thanks Pablo!

Just one question, is it `1.8.0` and not `v1.8.0` on purpose?

**Comment by odacremolbap** [comment]
> Just one question, is it `1.8.0` and not `v1.8.0` on purpose?

not on purpose, thanks

**Pull request merged**